libs/orbit-essentials/src/install_chunked_code.rs

```diff
@@ -26,27 +36,77 @@ struct EncodedAsset {
     pub sha256: Option<Vec<u8>>,
 }
 
+#[derive(Clone, Debug, CandidType, Deserialize)]
+struct GetChunkResponse {
+    #[serde(with = "serde_bytes")]
+    pub content: Vec<u8>,
+}
+
+// fetches extra chunks from an asset canister
+async fn fetch_extra_chunks(
+    store_canister: Principal,
+    extra_chunks_key: String,
+) -> Result<Vec<u8>, String> {
+    let asset = call::<_, (EncodedAsset,)>(
+        store_canister,
+        "get",
+        (GetArg {
+            key: extra_chunks_key.clone(),
+            accept_encodings: vec!["identity".to_string()],
+        },),
+    )
+    .await
+    .map_err(|(_, err)| format!("failed to fetch asset: {err}"))?
+    .0;
+    let mut res = asset.content;
+    let mut idx = 1_u64;
+    while res.len() < asset.total_length {
```

Comment:
let's prevent an asset canister from creating an infinite loop, ideally we can allow max number of iterations and same for max_length, if > then what the canister accepts (iirc 100mib) then we can directly fail this even before the loop.

Reply:
Fixed: https://github.com/dfinity/orbit/pull/392/commits/c15fc12d35bab66069f653ad59e0dadbf770f3d9